src/azure-cli/azure/cli/command_modules/cdn/custom/custom.py

```diff
@@ -84,13 +106,8 @@ def delete_profile(client, resource_group_name, profile_name):
     profile = None
     try:
         profile = client.profiles.get(resource_group_name, profile_name)
-    except ErrorResponseException as e:
-        props = getattr(e.inner_exception, 'additional_properties', {})
-        if not isinstance(props, dict) or not isinstance(props.get('error'), dict):
-            raise e
-        props = props['error']
-        if props.get('code') != 'ResourceNotFound':
-            raise e
+    except ResourceNotFoundError:
```

Comment:
From track 2 SDK, I think you'll always get `HttpResponseError` instead of `ResourceNotFoundError`.